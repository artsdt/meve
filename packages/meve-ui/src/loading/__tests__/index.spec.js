import Loading from '..'
import { mount } from '@vue/test-utils'

test('test loading to be false', () => {
  const wrapper = mount(Loading, {
    propsData: {
      loading: false,
    },
  })

  expect(wrapper.find('.m-loading__circle').isVisible()).toBe(false)
  expect(wrapper.find('.m-loading__description-container').isVisible()).toBe(false)

  wrapper.destroy()
})

test('test loading to be true', () => {
  const wrapper = mount(Loading, {
    propsData: {
      loading: true,
    },
  })

  expect(wrapper.find('.m-loading__circle').isVisible()).toBe(true)
  expect(wrapper.find('.m-loading__description-container').isVisible()).toBe(true)

  wrapper.destroy()
})

test('test loading styles', () => {
  const wrapper = mount(Loading, {
    propsData: {
      rotate: false,
      loading: true,
      description: 'description text',
      size: 40,
      color: '#000',
    },
  })

  expect(wrapper.html()).toMatchSnapshot()
  wrapper.destroy()
})

test('test loading slots', () => {
  const wrapper = mount(Loading, {
    propsData: {
      loading: true,
    },
    scopedSlots: {
      default: () => 'default slot',
      icon: () => 'icon slot',
      description: () => 'description slot',
    },
  })

  expect(wrapper.html()).toMatchSnapshot()
  wrapper.destroy()
})
