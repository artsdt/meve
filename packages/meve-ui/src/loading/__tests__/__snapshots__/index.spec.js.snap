// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test loading slots 1`] = `
"<div class=\\"m-loading m--box\\">default slot<div class=\\"m-loading__mask-container m-loading--overlay\\">
    <div class=\\"m-loading__mask m-loading--opacity-visible\\"></div>
    <div class=\\"m-loading__circle m-loading--rotate\\">icon slot</div>
    <div class=\\"m-loading__description-container\\">description slot</div>
  </div>
</div>"
`;

exports[`test loading styles 1`] = `
"<div class=\\"m-loading m--box\\">
  <div class=\\"m-loading__mask-container\\">
    <div class=\\"m-loading__mask m-loading--opacity-hidden\\"></div>
    <div class=\\"m-loading__circle\\"><svg viewBox=\\"25 25 50 50\\" style=\\"width: 40px; height: 40px; color: rgb(0, 0, 0);\\">
        <circle cx=\\"50\\" cy=\\"50\\" r=\\"20\\" fill=\\"none\\"></circle>
      </svg></div>
    <div class=\\"m-loading__description-container\\" style=\\"color: rgb(0, 0, 0);\\">
      <div class=\\"m-loading__description\\">description text</div>
    </div>
  </div>
</div>"
`;
